pkg/operator/latencyprofilecontroller/controller.go

```diff
@@ -0,0 +1,207 @@
+		if err != nil {
+			return err
+		}
+
+		// Collect the unique set of revisions of the node static pods
+		revisionMap := map[int32]struct{}{}
+		uniqueRevisions := []int32{}
+		for _, nodeStatus := range operatorStatus.NodeStatuses {
+			revision := nodeStatus.CurrentRevision
+			if _, ok := revisionMap[revision]; !ok {
+				revisionMap[revision] = struct{}{}
+				uniqueRevisions = append(uniqueRevisions, revision)
+			}
+		}
+
+		// Collect the current latency profile
+		configNodeObj, err := c.nodeLister.Get("cluster")
+		if err != nil && !errors.IsNotFound(err) {
+			return err
+		} else if errors.IsNotFound(err) {
+			// if config/v1/node/cluster object is not found this controller should do nothing
+			return nil
+		}
+
+		desiredControllerManagerArgumentVals := map[string]string{}
+		switch configNodeObj.Spec.WorkerLatencyProfile {
+		case apiconfigv1.DefaultUpdateDefaultReaction:
+			desiredControllerManagerArgumentVals[nodeMonitorGracePeriodArgument] = strconv.Itoa(int(apiconfigv1.DefaultNodeMonitorGracePeriod))
+		case apiconfigv1.MediumUpdateAverageReaction:
+			desiredControllerManagerArgumentVals[nodeMonitorGracePeriodArgument] = strconv.Itoa(int(apiconfigv1.MediumNodeMonitorGracePeriod))
+		case apiconfigv1.LowUpdateSlowReaction:
+			desiredControllerManagerArgumentVals[nodeMonitorGracePeriodArgument] = strconv.Itoa(int(apiconfigv1.LowNodeMonitorGracePeriod))
+		}
+
+		condition := metav1.Condition{
+			Type:   apiconfigv1.KubeControllerManagerDegraded,
+			Status: metav1.ConditionUnknown,
+		}
+		// do nothing if workerLatencyProfile is not set
+		if configNodeObj.Spec.WorkerLatencyProfile == "" {
+			// TODO: in case worker latency profile is transitioned
+			// from Default/Medium/Low to ""
+			// check that the required extendedArgs should vanish
+
+			condition.Reason = reasonWorkerLatencyProfileEmpty
+			condition.Message = "worker latency profile not set on cluster"
+			_, err := c.updateConfigNodeStatus(condition)
+			return err
+		}
+
+		// For each revision, check that the configmap for that revision have
+		// correct argument values or not
+		revisionsHaveSynced := true
+		for _, revision := range uniqueRevisions {
+			configMapNameWithRevision := fmt.Sprintf("%s-%d", controllerManagerConfigMapName, revision)
+			configMap, err := c.configMapClient.ConfigMaps(operatorclient.TargetNamespace).Get(ctx, configMapNameWithRevision, metav1.GetOptions{})
+			if err != nil {
+				return err
+			}
+			matches := true
+			for arg, val := range desiredControllerManagerArgumentVals {
+				match, err := configMatchesControllerManagerArgument(configMap, arg, val)
+				if err != nil {
+					return err
+				}
+				matches = matches && match
+			}
+			if !matches {
+				revisionsHaveSynced = false
+				break
+			}
+		}
+
+		condition.Status = metav1.ConditionTrue
+		condition.Reason = reasonWorkerLatencyProfileUpdated
+		if revisionsHaveSynced {
+			condition.Type = apiconfigv1.KubeControllerManagerComplete
```

Comment:
please take a look again at https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#typical-status-properties. This controller/operator should advertise all 3 conditions (types) simultaneously for easier consumption and overview of the current state.

The Status should oscillate between Unknown/True/False

Reply:
and better to do it in just one update call at the end

Reply:
Modified the logic to publish all 3 conditions during call to updateStatus.